hazelcast/src/main/java/com/hazelcast/internal/management/ManagementCenterService.java

```diff
@@ -272,6 +285,17 @@ public ConsoleCommandHandler getCommandHandler() {
         return commandHandler;
     }
 
+    /**
+     * Logs an event to Management Center.
+     *
+     * Events are used by Management Center to show the user what happens when on a cluster member.
+     */
+    public void log(Event event) {
+        if (this.managementCenterConfig.isEnabled()) {
+            events.add(event);
```

Comment:
This throws an exception in case the queue is full. We maybe don't want to cause change of control flow for the operation that logs the event. Isn't is better to replace the oldest entry instead? We do this in MC so it seams the most fitting solution here as well.

Reply:
There's no bound to the queue size (so it's the default value of `Integer.MAX_VALUE`), which I don't think will be hit since it's drained once every second. In case MC is not there to receive the events, queue is still drained so it doesn't grow indefinitely.

Now I had another look and realize that if the `ManagementCenterService` gets shutdown, events are still accepted into the queue, which might grow indefinitely. Will address that.

Reply:
Done.

Reply:
Isn't there also the case where the MC service is running but there is no MC? would that drain the queue anyway?

Reply:
>  In case MC is not there to receive the events, queue is still drained so it doesn't grow indefinitely.

yes, in that case, only the send fails, events are still drained: https://github.com/hazelcast/hazelcast/pull/13944/files#diff-3850ee7b4bd9df3672e6c4b18d1ec70cR372